### PR TITLE
chore(turborepo): add Nitro's env var pattern

### LINKED ISSUE #8504: Add env wildcards for framework Nitro NITRO_*
### Verify canary release

- [X] I verified that the issue exists in the latest Turborepo canary release.

### Link to code that reproduces this issue

-

### What package manager are you using / does the bug impact?

npm

### What operating system are you using?

Mac

### Which canary version will you have in your reproduction?

2.0.4-canary.4

### Describe the Bug

[Nitro](https://nitro.unjs.io) is the server framework that [Nuxt](https://nuxt.com) is using. It is part of every Nuxt application but is also an awesome API framework on its own that deploys everywhere (e.g. on Vercel Functions and Vercel Edge Functions).

When using Nitro separate all env variables start with `NITRO_*`. This wildcard should be added to the [Framework Inference](https://turbo.build/repo/docs/crafting-your-repository/using-environment-variables#framework-inference).

This will also fix deployments of Nuxt that deploy to any given preset that is not `node-server` (e.g. `vercel`), see https://github.com/vercel/turbo/issues/8503

### Expected Behavior

-

### To Reproduce

-

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by samydoesit**
### Description

Nitro pattern uses `NITRO_*`.
Reference: [nitro.unjs.io/config#runtimeconfig](https://nitro.unjs.io/config#runtimeconfig)

fixes #8504

### CODE REVIEW

**Comment by anthonyshew** [comment]
Thank you for this PR!

I'm going to have to ask a few questions for my own understanding. I've never used Nitro (or Nuxt beyond a `hello-world`) before so I want to make sure I'm understanding correctly:
- I see you're using `Strategy::Some`. Do you install _one of_ `nitropack` or `nitropack-nightly` to use Nitropack? (I'm figuring yes, but want to make sure I don't assume too much.)
- I'm confused by the intersection of Nuxt and Nitropack. If I were using Nuxt, do I need both the `NUXT_*` and `NITRO_*` namespaces? Are there situations where I'd use Nuxt and _not_ want/need the `NITRO_*` namespace?
  - If it's the case that Nuxt needs the `NITRO_*` namespace as well, could you add that wildcard to the Nuxt `env_wildcards`? That way, we would cover both the Nitro standalone and Nuxt use cases.

**Comment by samydoesit** [comment]
Hi 🙂

Question 1:
Yes, you need to install either `nitropack` or `nitropack-nightly` to use Nitropack.

Question 2:
When using Nuxt 3, Nitropack serves as the server engine. You can use both `NUXT_*` and `NITRO_*` namespaces. The `NUXT_*` namespace is specific to Nuxt's configuration and environment variables, while `NITRO_*` is specific to Nitropack's runtime and deployment settings.

I have also added it to Nuxt, as both Nuxt and Nitropack configurations can be used with Nuxt 3.

Here are some links for more information:
- [Nuxt Runtime Configuration](https://nuxt.com/docs/guide/going-further/runtime-config#example)
- [Nuxt/Nitro Deployment Preset](https://nuxt.com/docs/getting-started/deployment#presets)
- [Nitro Environment Variables](https://nitro.unjs.io/deploy/runtimes/node#environment-variables)

File: docs/repo-docs/crafting-your-repository/using-environment-variables.mdx [hunk 1]
```diff
@@ -63,7 +63,8 @@ Turborepo automatically adds prefix wildcards to your [`env`](/repo/docs/referen
 | Create React App | `REACT_APP_*`     |
 | Gatsby           | `GATSBY_*`        |
 | Next.js          | `NEXT_PUBLIC_*`   |
-| Nuxt.js          | `NUXT_*`      |
+| Nitro            | `NITRO_*`         |
+| Nuxt.js          | `NUXT_*`          |
```
Comment (anthonyshew) [inline docs/repo-docs/crafting-your-repository/using-environment-variables.mdx:67]:
```suggestion
| Nuxt.js          | `NUXT_*`, `NITRO_*`          |
```

**Comment by anthonyshew** [comment]
Great! Thank you for the documentation links, as well.

Just have my one suggestion to update the doc and then we can run tests and merge!

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 13d77bd6025025accfd111de2ae543d42c088af0

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by anthonyshew** [comment]
@samydoesit Looks like we need to bump the vector size. I think we'll be able to ship after that!
- [Error from GitHub Action](https://github.com/vercel/turbo/actions/runs/9909323042/job/27377147521?pr=8725#step:6:920)
- [Line you'll update](https://github.com/vercel/turbo/pull/8725/files#diff-0ed3e867abaf0441136e53afa2b3318d5c984803b810b20fdbe0f0f555c097d7R34)

**Comment by samydoesit** [comment]
I've increased the vector size.
Thank you for your support and review!

**Review by anthonyshew — APPROVED** [review]
Great! Thanks!

**Comment by MickL** [comment]
Amazing, thanks for the PR :)
The update might be added to [the docs](https://turbo.build/repo/docs/crafting-your-repository/using-environment-variables#framework-inference) too

**Pull request merged**